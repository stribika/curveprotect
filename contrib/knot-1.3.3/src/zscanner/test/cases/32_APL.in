$ORIGIN	.
$TTL	1

; OK
@	APL				; The simplest variant - blank list
@	APL	1:0.0.0.0/0		; Minimal ipv4 prefix length
@	APL	1:255.255.255.255/32	; Maximal ipv4 prefix length
@	APL	1:255.255.255.255/30	; Prefix length isn't multiple of 8
@	APL	2:0::0/0		; Minimal ipv6 prefix length
@	APL	2:0::0/128		; Maximal ipv6 prefix length
@	APL	2:FFFF:FFFF:FFFF::/2	; Trailing zeroes test
@	APL	!1:0.0.0.0/0		; Negation flag
@	APL	1:0.0.0.0/0 1:255.255.255.255/32	; More APLs
@	TYPE42	\# 4 00010000		; TYPE + Hexadecimal rdata
@	TYPE42	1:0.0.0.0/0		; TYPE
@	APL	\# 0			; Zero length rdata
@	apl	1:0.0.0.0/0		; Type in lower-case

; KO
@	APL	0:0.0.0.0/32		; Bad address family
@	APL	x:0.0.0.0/32		; Bad address family
@	APL	!x:0.0.0.0/32		; Bad address family
@	APL	1:0.0.0.0/33		; Prefix length is too long
@	APL	2:0::0/129		; Prefix length is too long
@	APL	2::/0			; Bad ipv6 address
@	APL	2:0::0/x		; Bad prefix length
@	APL	1:0.0.0.0/		; Missing prefix length
@	APL	\#			; Missing hex length
@	APL	\# 4 0001000000		; Too long rdata
@	APL	\# 5 00000000		; Bad rdata length
