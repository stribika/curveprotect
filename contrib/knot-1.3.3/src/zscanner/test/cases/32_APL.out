OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00010000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00012004FFFFFFFF
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00011E04FFFFFFFF
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00020000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00028000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00020206FFFFFFFFFFFF
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00010080
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=0001000000012004FFFFFFFF
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00010000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00010000
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=
------
OWNER=00
CLASS=0001
RRTTL=00000001
RTYPE=002A
RDATA=00010000
------
WARNG=ZSCANNER_EBAD_APL
------
WARNG=ZSCANNER_EBAD_APL
------
WARNG=ZSCANNER_EBAD_APL
------
WARNG=ZSCANNER_EBAD_APL
------
WARNG=ZSCANNER_EBAD_APL
------
WARNG=ZSCANNER_EBAD_IPV6
------
WARNG=ZSCANNER_EBAD_APL
------
WARNG=ZSCANNER_EBAD_APL
------
WARNG=ZSCANNER_EBAD_NUMBER
------
WARNG=ZSCANNER_EBAD_RDATA_LENGTH
------
WARNG=ZSCANNER_EBAD_RDATA_LENGTH
------
